policy/policy_handler/async.go

```diff
@@ -84,6 +89,13 @@ func buildAsyncDataSources(multipleQuerySupport bool) dataSourceProvider {
 		if err != nil {
 			return nil, fmt.Errorf("failed to unmarshal async query result: %w", err)
 		}
+		if len(queryResponse.Errors) > 0 {
+			errorMessage := queryResponse.Errors[0].Message
+			if errorMessage == "An unexpected error has occurred" {
+				return nil, fmt.Errorf("async query contained error: %s", errorMessage).(types.RetryableExecutionError)
+			}
+			return nil, fmt.Errorf("async query contained error: %s", errorMessage)
+		}
```

Comment:
I'm good with this being here, but you should also then remove the error handling in gql_async.go

Reply:
Good shout.